File: src/main/java/no/nav/bidrag/arbeidsflyt/BidragArbeidsflyt.java
package no.nav.bidrag.arbeidsflyt;

import com.fasterxml.jackson.databind.ObjectMapper;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Bean;

@SpringBootApplication
public class BidragArbeidsflyt {

  @Bean
  public ObjectMapper objectMapper() {
    return new ObjectMapper();
  }

  public static void main(String... args) {
    SpringApplication.run(BidragArbeidsflyt.class, args);
  }
}

File: src/main/kotlin/no/nav/bidrag/arbeidsflyt/BidragArbeidsflyt.kt
package no.nav.bidrag.arbeidsflyt

import no.nav.security.token.support.spring.api.EnableJwtTokenValidation
import org.slf4j.LoggerFactory
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.security.autoconfigure.SecurityAutoConfiguration
import org.springframework.boot.security.autoconfigure.UserDetailsServiceAutoConfiguration
import org.springframework.boot.security.autoconfigure.actuate.web.servlet.ManagementWebSecurityAutoConfiguration
import org.springframework.boot.security.autoconfigure.web.servlet.ServletWebSecurityAutoConfiguration
import org.springframework.context.annotation.EnableAspectJAutoProxy

const val PROFILE_NAIS = "nais"
const val PROFILE_KAFKA_TEST = "kafka_test"
val SECURE_LOGGER = LoggerFactory.getLogger("secureLogger")

@SpringBootApplication(
    exclude = [
        SecurityAutoConfiguration::class,
        ManagementWebSecurityAutoConfiguration::class,
        UserDetailsServiceAutoConfiguration::class,
        ServletWebSecurityAutoConfiguration::class,
    ],
)
@EnableJwtTokenValidation(ignore = ["org.springframework", "org.springdoc"])
@EnableAspectJAutoProxy
class BidragArbeidsflyt

fun main(args: Array<String>) {
    val app = SpringApplication(BidragArbeidsflyt::class.java)
    app.setAdditionalProfiles(if (args.isEmpty()) PROFILE_NAIS else args[0])
    app.run(*args)
}
